app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java

```diff
@@ -511,4 +543,57 @@ public Snackbar showSnackbarAbovePlayer(CharSequence text, int duration) {
     public Snackbar showSnackbarAbovePlayer(int text, int duration) {
         return showSnackbarAbovePlayer(getResources().getText(text), duration);
     }
+
+    /**
+     * Handles the deep link incoming via App Actions.
+     * Performs an in-app search or opens the relevant feature of the app
+     * depending on the query.
+     *
+     * @param uri incoming deep link
+     */
+    private void handleDeeplink(Uri uri) {
+        if (uri == null || uri.getPath() == null) {
+           return;
+        }
+
+        switch (uri.getPath()) {
+            case "/search":
+                String query = uri.getQueryParameter("query");
+                if (query == null) {
+                    return;
+                }
+
+                this.loadChildFragment(OnlineSearchFragment.newInstance(CombinedSearcher.class, query));
+                break;
+
+            case "/main":
+                String feature = uri.getQueryParameter("featureName");
+                if (feature == null) {
+                    return;
+                }
+
+                feature = feature.toLowerCase();
+                switch (feature) {
+                    case "downloads":
```

Comment:
This looks pretty similar to the `switch` in `loadFragment`. Could that one be re-used instead of creating another switch?

Reply:
The feature names (and therefore, the parameters used in a potential user query in the Assistant) don’t exactly match the Strings in the loadFragment switch so it’s not possible to simply re-use it. One solution would be to add an additional method to convert the strings first, which would then get rid of the second switch.

Reply:
Is this the same as the identifier specified in `actions.xml` (`entity identifier="DOWNLOADS"`)? If so, why do you use lower case here and upper case in `actions.xml`?